### PR TITLE
Create Shared Link

### PR DESCRIPTION
**Pull request by marek-vysoky**
Adding comments, adding effective_access property

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

**Comment by marek-vysoky** [comment]
I've updated comments for CreateSharedLinkAsync. Can you check again?

**Comment by cburnette** [comment]
@coolboy Let's review this one.

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 1]
```diff
@@ -173,11 +173,20 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
         }
 
         /// <summary>
-        /// Used to create a shared link for this particular folder. Please see here for more information on the 
+        /// Used to create a shared link for this particular folder. Please see <see href="https://community.box.com/t5/Collaboration-and-Sharing/Shared-Links-Overview-And-FAQs/ta-p/142">here</see> for more information on the
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: should we use this link? https://docs.box.com/reference#create-a-shared-link-for-a-folder

**Comment by coolboy** [comment]
@cburnette I have put my comments, @marek-vysoky could you check?

**Comment by marek-vysoky** [comment]
@coolboy I've changed link. Can you check it?

**Comment by coolboy** [comment]
@marek-vysoky what my other comments regarding the optional, @cburnette

**Comment by coolboy** [comment]
sorry, accidentally click on the closed the PR, reopen

File: Box.V2/Models/Request/BoxSharedLinkRequest.cs [hunk 2]
```diff
@@ -33,6 +33,12 @@ public class BoxSharedLinkRequest
         /// </summary>
         [JsonProperty(PropertyName = "password")]
         public string Password { get; set; }
```
Comment (marek-vysoky) [inline Box.V2/Models/Request/BoxSharedLinkRequest.cs]: @coolboy  When I look to documentation, fields Password, Permissions are not required
https://docs.box.com/reference#create-a-shared-link-for-a-folder
Do you think that  I should check that fields are mandatory

**Comment by cburnette** [comment]
functionality previously moved into master

**Pull request closed without merging**